straxen/plugins/daqreader.py

```diff
@@ -156,8 +156,9 @@ def _load_chunk(self, path, start, end, kind='central'):
         first_start, last_start, last_end = None, None, None
         if len(records):
             first_start, last_start = records[0]['time'], records[-1]['time']
-            # Records are fixed length, so we also know the last end:
-            last_end = strax.endtime(records[-1])
+            # Records are sorted by (start)time and are of variable length.
+            # Their endtimes can differ.
+            last_end = strax.endtime(records).max()
```

Comment:
Maybe my logic is flawed, but would not it be enough for records to check the last len(data) - 1 * channel records since they are fixed length? We cannot have two record spanning the same time range for the same channel, hence we have only len(data) - 1  * channel possibilities. Maybe use 110 + 1 * 494  in case I did a mistake in my calculation :D

Example:
len(data) = 5 dashes 

ch4: |-----| 
ch3: _|-|    
ch2: __|-|
ch1: ___|-|
ch6: ___|-|
ch8: ___|--|
ch7: _____|-| <- same end time as ch4.
etc.

Reply:
Might be equivalent to testing all raw_records anyhow.

Reply:
Yeah thanks Daniel, indeed I think the worse case scenario is:
len(data) = 5 dashes
```
ch1: |-|-|-|-|_
ch2: .|-|-|-|-|_
ch3: ..|-----|
```
Sorted by start time all records of ch1 and ch2 precede ch3. So we would indeed need (len(data) -1 ) * n_channels to get the the record of channel 3.

Although the time needed is not that large, the relative times for one of the latest 200 MB/s runs for a given chunk is significant:
![afbeelding](https://user-images.githubusercontent.com/22295914/87038596-43144300-c1ee-11ea-85a0-116b528d4b0e.png)

I'll change it

Reply:
Yeah, I'm not sure that there's much difference between checking the last 50,000 records and the whole chunk

Reply:
Also, if I read those benchmarks correctly, it's faster to do the whole chunk than part of it, by a factor of nearly 3

Reply:
This would be the implementation I'd go for which is like 500 us. In the most pessimistic case we'd have to consider all known channels.

```
            # Records are sorted by (start)time and are of variable length.
            # Their end-times can differ. In the most pessimistic case we have
            # to look back one record length for each channel.
            tot_channels = np.sum([np.diff(x)+1 for x in
                                   self.config['channel_map'].values()])
            look_n_samples = self.config["record_length"] * tot_channels
            last_end = strax.endtime(records[-look_n_samples:]).max()
```